core/src/main/java/org/bitcoinj/store/SPVBlockStore.java

```diff
@@ -327,10 +378,24 @@ public void close() throws BlockStoreException {
     //   12 bytes of chain work
     //    4 bytes of height
     //   80 bytes of block header data
+
+    static final int RECORD_SIZE_V2 = 32 /* hash */ + StoredBlock.COMPACT_SERIALIZED_SIZE_V2;
+
+    // V2 file format:
+    //   4 magic header bytes = "SPV2"
+    //   4 cursor bytes, which indicate the offset from the first kb where the next block header should be written.
+    //   32 bytes for the hash of the chain head
+    //
+    // For each header (148 bytes)
+    //   32 bytes hash of the header
+    //   32 bytes of chain work
+    //    4 bytes of height
+    //   80 bytes of block header data
+
```

Comment:
I deliberately decided against this, because the v2 format is the relevant one in future and v1 documentation is only there for historic purposes. We could decide to remove the v1 doc though.

Reply:
But then again, yeah maybe you're right.

Reply:
See 38d7d6c786832310d4f8ac405632256e1df0ac3e.

Reply:
Look great! Resolved.